### DISCUSSION TITLE
Per-player entity display name

### DISCUSSION THREAD

**Discussion by ZakShearman** [Minestom/Minestom#706] [Help] [answered]
I'm trying to create an entity that has a different name based on a user's settings. I'd like to do so preferably without sending an unnecessary meta packet with invalid values when the entity is created.

I have tried overriding `updateNewViewer`; however, `viewEngine` has protected access so I cannot properly override the method.

Additionally, I cannot override `getMetadataPacket` as this does not provide me with the viewer.

**Comment by TheMode** [comment, marked as answer]
I do not think I understand what is preventing you from overriding `updateNewViewer`? As `viewEngine` is unused there, and you should have access to the protected `metadata` field.

Ideally you would want to only transform the packet you are interested in, which is not possible but planned (using `PlayerPacketOutEvent`)

edit: I have found the mentioned usage of `viewEngine`, and it has been removed. It however does not change my answer.

**Reply by ZakShearman** [reply]
So the course of action should be to wait until PlayerPacketOutEvent has been implemented?

In the meantime, I could modify the metadata. Is there a good way for me to duplicate the `Metadata` object used in updateNewViewer or create my own two instances of `EntityMeta` that can be updated throughout the life of the entity (this would be preferred)?
